internal/fwserver/server_planresourcechange_test.go

```diff
@@ -3007,7 +3009,130 @@ func TestServerPlanResourceChange(t *testing.T) {
+					ModifyPlanMethod: func(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+						if req.ClientCapabilities.DeferralAllowed == true {
+							resp.Deferred = &resource.Deferred{Reason: resource.DeferredReasonAbsentPrereq}
+						}
+
+						var data testSchemaData
+
+						resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+						data.TestComputed = types.StringValue("test-plannedstate-value")
+
+						resp.Diagnostics.Append(resp.Plan.Set(ctx, &data)...)
+					},
+				},
+			},
+			expectedResponse: &fwserver.PlanResourceChangeResponse{
+				Deferred: &resource.Deferred{Reason: resource.DeferredReasonProviderConfigUnknown},
+				PlannedState: &tfsdk.State{
+					Raw: tftypes.NewValue(testSchemaType, map[string]tftypes.Value{
+						"test_computed": tftypes.NewValue(tftypes.String, "test-plannedstate-value"),
+						"test_required": tftypes.NewValue(tftypes.String, "test-config-value"),
+					}),
+					Schema: testSchema,
+				},
+				PlannedPrivate: testEmptyPrivate,
+			},
+		},
+		"create-resourcewithmodifyplan-response-deferral-automatic-skip-plan-modification": {
+			server: &fwserver.Server{
+				Provider: &testprovider.Provider{
+					SchemaMethod: func(_ context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {},
+					ConfigureMethod: func(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+						resp.Deferred = &provider.Deferred{Reason: provider.DeferredReasonProviderConfigUnknown}
+					},
+				},
+			},
+			configureProviderReq: &provider.ConfigureRequest{
+				ClientCapabilities: provider.ConfigureProviderClientCapabilities{
+					DeferralAllowed: true,
+				},
+			},
+			request: &fwserver.PlanResourceChangeRequest{
+				Config: &tfsdk.Config{
+					Raw: tftypes.NewValue(testSchemaType, map[string]tftypes.Value{
+						"test_computed": tftypes.NewValue(tftypes.String, nil),
+						"test_required": tftypes.NewValue(tftypes.String, "test-config-value"),
+					}),
+					Schema: testSchema,
+				},
+				ProposedNewState: &tfsdk.Plan{
+					Raw: tftypes.NewValue(testSchemaType, map[string]tftypes.Value{
+						"test_computed": tftypes.NewValue(tftypes.String, nil),
+						"test_required": tftypes.NewValue(tftypes.String, "test-config-value"),
+					}),
+					Schema: testSchema,
+				},
+				PriorState:     testEmptyState,
+				ResourceSchema: testSchema,
+				Resource: &testprovider.ResourceWithModifyPlan{
+					ModifyPlanMethod: func(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
+						if req.ClientCapabilities.DeferralAllowed == true {
+							resp.Deferred = &resource.Deferred{Reason: resource.DeferredReasonAbsentPrereq}
+						}
+
+						var data testSchemaData
+
+						resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+						data.TestComputed = types.StringValue("test-plannedstate-value")
+
+						resp.Diagnostics.Append(resp.Plan.Set(ctx, &data)...)
```

Comment:
Similar though to the import test, let's return a diagnostic here so the test fails if this is called